### PR TITLE
Remove godot-headers submodule, copy files directly

### PR DESCRIPTION
**Pull request by akien-mga**
With the new GDExtension API, the headers are just two files, one of
which is generated and needs to always be kept in sync with the Godot
engine version.

So there's little practical use for using godot-headers as a submodule
anymore, and it only makes godot-cpp updates more cumbersome.

Custom headers (i.e. a custom API JSON) can still be used by passing
the `headers_dir` SCons option.

### CODE REVIEW

File: godot-headers/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# godot-headers
+
+This repository contains C headers for
+[**Godot Engine**](https://github.com/godotengine/godot)'s *GDNative Extensions* API.
```
Comment (akien-mga) [inline godot-headers/README.md:4]: I kept the README mostly for the "Updating Headers" instruction. I could cut it down to only that part to remove the references to the headers repo, `3.x` branch, etc. But I figured let's keep it simple for now.

File: godot-headers/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+Godot version.**
+
+## Using this library
+
+This library is meant to be used in conjunction with a code generator that creates the binding code necessary to interact with Godot. This library only provides access to the API over which communication with the Godot core is implemented and binding meta data in the form of a json document.
+
+One such binding library is [godot-cpp](https://github.com/godotengine/godot-cpp) which implements the bindings using C++.
+
+A pure C sample project is currently not available.
+
```
Comment (bruvzg) [inline godot-headers/README.md:41]: All of this seems irrelevant, probably should just keep `Updating Headers`.

File: godot-headers/LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# MIT License
+
+Copyright (c) 2017-2022 Godot Engine contributors.
+
```
Comment (bruvzg) [inline godot-headers/LICENSE.md:4]: Since the license is identical to the `godot-cpp` license, probably can be removed as well.

File: godot-headers/README.md (hunk 2 above)
Comment (akien-mga) [reply godot-headers/README.md]: Fair enough, let's keep it short.
